### PR TITLE
FOCUS #330: Replace ChargeSubcategory references after ChargeClass inclusion

### PR DESCRIPTION
**Pull request by flanakin**

### CODE REVIEW

File: specification/columns/pricingquantity.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
...
 
-The PricingQuantity column MUST be present in the billing data. This column MUST be of type Decimal and MUST conform to [Numeric Format](#numericformat) requirements. The value MAY be negative in cases where [ChargeSubcategory](#chargesubcategory) is "Refund". This column MUST NOT be null if [ChargeCategory](#chargecategory) is "Usage" or "Purchase" and MAY be null for other ChargeCategory values. When unit prices are not null, multiplying PricingQuantity by a unit price MUST produce a result equal to the corresponding cost metric.
+The PricingQuantity column MUST be present in the billing data. This column MUST be of type Decimal and MUST conform to [Numeric Format](#numericformat) requirements. The value MAY be negative in cases where [ChargeClass](#chargeclass) is "Correction". This column MUST NOT be null if [ChargeCategory](#chargecategory) is "Usage" or "Purchase" and MAY be null for other ChargeCategory values. When unit prices are not null, multiplying PricingQuantity by a unit price MUST produce a result equal to the corresponding cost metric.
```
Comment (aqu-erp) [inline specification/columns/pricingquantity.md:5]:
```suggestion
The PricingQuantity column MUST be present in the billing data. This column MUST be of type Decimal and MUST conform to [Numeric Format](#numericformat) requirements. The value MAY be negative in cases where [ChargeClass](#chargeclass) is "Correction". This column MUST NOT be null if [ChargeCategory](#chargecategory) is "Usage" or "Purchase" and [ChargeClass](#chargeclass) is "Regular". This column MAY be null for other ChargeCategory values. When unit prices are not null, multiplying PricingQuantity by a unit price MUST produce a result equal to the corresponding cost metric.
```

Do we need to raise a separate issue to review all instances of `PricingCategory MUST NOT be null if [ChargeCategory] is "Usage" or "Purchase"` impacted by bulk correction concept?

**Review by aqu-erp — CHANGES_REQUESTED** [review]

File: specification/columns/pricingquantity.md (hunk 1 above)
Comment (ijurica) [reply specification/columns/pricingquantity.md:5]: We discussed this at the TF-1 meeting on 4/16/2024 (see https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/issues/366). 
Agreed to create a separate PR.

**Comment by ijurica** [comment]
Current version of the ChargeCategory Description requires updating.

> Indicates whether the row represents an upfront or recurring fee, cost of usage that already occurred, an after-the-fact adjustment (e.g., credits), or taxes.

@flanakin Could you please update that as well? Closely resemble the introductory sentence in the column definition?1

File: specification/attributes/discount_handling.md [hunk 2]
```diff
@@ -38,7 +38,7 @@ Indicates how to include and apply discounts to usage charges or rows.
   * ChargeSubcategory MUST be "Used Commitment" for rows that received a reduced price from that commitment.
   * If a commitment is not fully utilized, the provider MUST include a row that represents the unused portion of the commitment for that charge period. ChargeSubcategory MUST be "Unused Commitment".
   * The sum of the EffectiveCost for all "Used Commitment" and "Unused Commitment" rows for each CommitmentDiscountId over the entire duration of the commitment MUST be the same as the total BilledCost of the commitment-based discount.
-* Credits that are applied after the fact MUST use a ChargeType of "Adjustment" and ChargeSubcategory of "Credit".
+* Credits that are applied after the fact MUST use a ChargeType of "Adjustment" and ChargeCategory of "Credit".
```
[the remark this answers is not in the document]
Comment (udam-f2) [reply specification/attributes/discount_handling.md:41]: Line 34 above also references ChargeSubCategory

**Comment by jpradocueva** [comment]
Approved by the group on April 18

**Comment by udam-f2** [comment]
> Approved by the group on April 18

There are open conversations on this PR that should be resolved before merging.

**Review by thecloudman — APPROVED** [review]
Suggest we open another PR to change "Regular" in ChargeClass to "Standard" to match changes in PR383 for PricingCategory where "On-Demand" was changed to "Standard". All other references look good. @flanakin

**Comment by ahullah** [comment]
@thecloudman have approved that separate PR #410

**Comment by ijurica** [comment]
Reminders:

- **ChargeClass** column spec still contains - `Column required` constraint instead of `Feature level`
- Could you please replace the `Requirement level: Mandatory - Optional` with `Feature level: Mandatory - Conditional` in **Grouping constructs Appendix** as well.

File: specification/columns/contractedcost.md [hunk 3]
```diff
@@ -23,14 +23,14 @@ Cost calculated by multiplying *contracted unit price* and the corresponding Pri
...
-| Value format    | [Numeric Format](#numericformat) |
-| Number range    | Any valid decimal value |
+| Constraint    | Value                            |
+| :------------ | :------------------------------- |
+| Column type   | Metric                           |
+| Feature level | Mandatory                        |
+| Allows nulls  | False                            |
+| Data type     | Decimal                          |
+| Value format  | [Numeric Format](#numericformat) |
+| Number range  | Any valid decimal value          |
```
Comment (flanakin) [inline specification/columns/contractedcost.md:33]:
```suggestion
| Constraint      | Value                   |
|:----------------|:------------------------|
| Column type     | Metric                  |
| Feature level   | Mandatory               |
| Allows nulls    | False                   |
| Data type       | Decimal                 |
| Value format    | [Numeric Format](#numericformat) |
| Number range    | Any valid decimal value |```

File: specification/columns/effectivecost.md [hunk 4]
```diff
@@ -36,14 +36,14 @@ Eligible purchases should be *amortized* using a methodology determined by the p
...
-| Value format    | [Numeric Format](#numericformat) |
-| Number range    | Any valid decimal value |
+| Constraint    | Value                            |
+| :------------ | :------------------------------- |
+| Column type   | Metric                           |
+| Feature level | Mandatory                        |
+| Allows nulls  | False                            |
+| Data type     | Decimal                          |
+| Value format  | [Numeric Format](#numericformat) |
+| Number range  | Any valid decimal value          |
```
Comment (flanakin) [inline specification/columns/effectivecost.md:46]:
```suggestion
|    Constraint   |      Value              |
|:----------------|:------------------------|
| Column type     | Metric                  |
| Feature level   | Mandatory               |
| Allows nulls    | False                   |
| Data type       | Decimal                 |
| Value format    | [Numeric Format](#numericformat) |
| Number range    | Any valid decimal value |
```

File: specification/columns/listcost.md [hunk 5]
```diff
@@ -23,14 +23,14 @@ Cost calculated by multiplying List Unit Price and the corresponding Pricing Qua
...
-| Value format    | [Numeric Format](#numericformat) |
-| Number range    | Any valid decimal value |
+| Constraint    | Value                            |
+| :------------ | :------------------------------- |
+| Column type   | Metric                           |
+| Feature level | Mandatory                        |
+| Allows nulls  | False                            |
+| Data type     | Decimal                          |
+| Value format  | [Numeric Format](#numericformat) |
+| Number range  | Any valid decimal value          |
```
Comment (flanakin) [inline specification/columns/listcost.md:33]:
```suggestion
| Constraint      | Value                   |
|:----------------|:------------------------|
| Column type     | Metric                  |
| Feature level   | Mandatory               |
| Allows nulls    | False                   |
| Data type       | Decimal                 |
| Value format    | [Numeric Format](#numericformat) |
| Number range    | Any valid decimal value |
```

File: specification/columns/pricingquantity.md [hunk 6]
```diff
@@ -18,14 +18,14 @@ The volume of a given SKU associated with a *resource* or *service* used or purc
...
-| Value format    | [Numeric Format](#numericformat) |
-| Number Range    | Any valid decimal value   |
+| Constraint    | Value                            |
+| :------------ | :------------------------------- |
+| Column type   | Metric                           |
+| Feature level | Mandatory                        |
+| Allows nulls  | True                             |
+| Data type     | Decimal                          |
+| Value format  | [Numeric Format](#numericformat) |
+| Number Range  | Any valid decimal value          |
```
Comment (flanakin) [inline specification/columns/pricingquantity.md:28]:
```suggestion
|    Constraint   |      Value                |
|:----------------|:--------------------------|
| Column type     | Metric                    |
| Feature level   | Mandatory                 |
| Allows nulls    | True                      |
| Data type       | Decimal                   |
| Value format    | [Numeric Format](#numericformat) |
| Number Range    | Any valid decimal value   |
```

File: specification/columns/usagequantity.md [hunk 7]
```diff
@@ -18,14 +18,14 @@ The volume of a given SKU associated with a *resource* or *service* used or purc
...
-| Value format    | [Numeric Format](#numericformat) |
-| Number range    | Any valid decimal value |
+| Constraint    | Value                            |
+| :------------ | :------------------------------- |
+| Column type   | Metric                           |
+| Feature level | Conditional                      |
+| Allows nulls  | True                             |
+| Data type     | Decimal                          |
+| Value format  | [Numeric Format](#numericformat) |
+| Number range  | Any valid decimal value          |
```
Comment (flanakin) [inline specification/columns/usagequantity.md:28]:
```suggestion
| Constraint      | Value         |
|:----------------|:--------------|
| Column type     | Metric        |
| Feature level   | Conditional   |
| Allows nulls    | True          |
| Data type       | Decimal       |
| Value format    | [Numeric Format](#numericformat) |
| Number range    | Any valid decimal value |
```

File: supporting_content/attributes/discount_handling.md [hunk 8]
```diff
@@ -12,20 +12,20 @@ For the sake of this example, let's say the negotiated and bundled discounts cap
...
+| Commitment only               | Foo2     | 20    | $1.00 | $1.00    | $0.00  | $0.50     |
+| Uncommitted part              | Foo2     | 40    | $2.00 | $2.00    | $2.00  | $2.00     |
+| Commitment only               | Foo3     | 10    | $0.50 | $0.50    | $0.00  | $0.25     |
+| Unused commitment             |          |       | $0.00 | $0.00    | $0.00  | $0.25     |
+| Negotiated +&nbsp;committed   | Foo4     | 22.22 | $1.11 | $1.00    | $0.00  | $0.50     |
+| Negotiated +&nbsp;uncommitted | Foo4     | 37.78 | $1.89 | $1.70    | $1.70  | $1.70     |
+| All discounts<sup>1</sup>     | Foo5     | 24.69 | $1.11 | $1.00    | $0.00  | $0.50     |
+| Negotiated +&nbsp;bundled     | Foo5     | 5.31  | $0.24 | $2.15    | $2.15  | $2.15     |
+| Negotiated only               | Foo5     | 15    | $0.75 | $0.67    | $0.67  | $0.67     |
+| All discounts capped out      | Foo5     | 15    | $0.75 | $0.75    | $0.75  | $0.75     |
```
Comment (flanakin) [inline supporting_content/attributes/discount_handling.md:28]:
```suggestion
| Scenario | Resource | Qty | List | OnDemand | Billed | Effective |
|-----------|-----------|------|-----|--------------|-------|-----------|
| No discounts | Foo0 | 60 | $3.00 | $3.00 | $3.00 | $3.00 |
| Negotiated only | Foo1 |  60 | $3.00 | $2.70 | $2.70 | $2.70 |
| Commitment only | Foo2 |  20 | $1.00 | $1.00 | $0.00 | $0.50 |
| Uncommitted part | Foo2 | 40 | $2.00 | $2.00 | $2.00 | $2.00 |
| Commitment only | Foo3 | 10 | $0.50 | $0.50 | $0.00 | $0.25 |
| Unused commitment | | | $0.00 | $0.00 | $0.00 | $0.25 |
| Negotiated +&nbsp;committed | Foo4 | 22.22 | $1.11 | $1.00 | $0.00 | $0.50 |
| Negotiated +&nbsp;uncommitted | Foo4 | 37.78 | $1.89 | $1.70 | $1.70 | $1.70 |
| All discounts<sup>1</sup> | Foo5 | 24.69 | $1.11 | $1.00 | $0.00 | $0.50 |
| Negotiated +&nbsp;bundled | Foo5 | 5.31 | $0.24 | $2.15 | $2.15 | $2.15 |
| Negotiated only | Foo5 | 15 | $0.75 | $0.67 | $0.67 | $0.67 |
| All discounts capped out | Foo5 | 15 | $0.75 | $0.75 | $0.75 | $0.75 |
```

File: supporting_content/columns/chargecategory.md [hunk 9]
```diff
@@ -8,111 +8,111 @@ Current column mappings found in available data sets:
...
+| Microsoft | Adjustment                         | Adjustment     | Rounding errors.                                                                                                                                                                                                                                                                                                                                           |
+| Microsoft | Tax                                | Tax            | US sales tax or VAT.                                                                                                                                                                                                                                                                                                                                       |
```
Comment (flanakin) [inline supporting_content/columns/chargecategory.md:38]:
```suggestion
| --------- | ---------------------------------- | ---------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| AWS       | Anniversary / Usage                | Usage      | Usage charged at on-demand rate for resources with resource id (EC2, EBS, RDS, RedShift)                                                                                                                                                                                                                                                                   |
| AWS       | Anniversary / Usage                | Usage      | Usage charged at on-demand rate for resources with "no resource id" (Support, CloudWatch, DataTransfer)                                                                                                                                                                                                                                                    |
| AWS       | Anniversary / Tax                  | Tax        | US sales tax or VAT with "no resource id"                                                                                                                                                                                                                                                                                                                  |
| AWS       | Purchase / Fee                     | Purchase   | All upfront and partial upfront fees for RI purchase                                                                                                                                                                                                                                                                                                       |
| AWS       | Purchase / RIFee                   | Purchase   | Monthly recurring RI amount for partial upfront and no upfront                                                                                                                                                                                                                                                                                             |
| AWS       | Purchase / SavingsPlanUpfrontFee   | Purchase   | All upfront and partial upfront fees for SP purchase                                                                                                                                                                                                                                                                                                       |
| AWS       | Purchase / SavingsPlanRecurringFee | Purchase   | Monthly recurring SP amount for partial upfront and no upfront                                                                                                                                                                                                                                                                                             |
| AWS       | Adjustments / SavingsPlanNegation  | Usage      | Used to negate the on-demand cost covered by SP                                                                                                                                                                                                                                                                                                            |
| AWS       | Anniversary / BundledDiscount      | Usage      | Usage based discount for free or discounted price. If a customer uses X units of product/service A, this customer gets Y units of product/service B at a discounted price (with a discount Z%).                                                                                                                                                            |
| AWS       | Refund                             | Adjustment |
| GCP       | regular                            | Usage      | These show up as rows that contain data of usage and costs                                                                                                                                                                                                                                                                                                 |
| GCP       | tax                                | Tax        | These show up as monthly rows without a project as a credit and with a project with a debit.                                                                                                                                                                                                                                                               |
| GCP       | adjustment                         | Adjustment | ![Screenshot of GCP cost details with type and mode columns.](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/assets/399533/af90e4cd-f3c0-448a-bb0f-0249bcf7135c)<br>Example 1:<br>Description: "Billing correction - Adjustment for project X for incorrect Flexible CUD charge"<br>Mode: "MANUAL_ADJUSTMENT"<br>Type: "GENERAL_ADJUSTMENT" |
| GCP       | rounding_error                     | Adjustment | These show up as monthly rows without a project as a credit                                                                                                                                                                                                                                                                                                |
| GCP       | credit                             | Adjustment | Fields: type, name, amount, full_name, id<br>![Screenshot of a table with a type column and 5 rows of example values](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/assets/399533/15bcc210-5a36-473b-aeac-c1d2682dfdc8)                                                                                                                    |
| Microsoft | Purchase                           | Purchase   | Upfront or recurring fee for Marketplace offers or commitment-based discounts.                                                                                                                                                                                                                                                                             |
| Microsoft | Usage                              | Usage      | Consumption-based usage of deployed resources.                                                                                                                                                                                                                                                                                                             |
| Microsoft | Refund                             | Adjustment | Refund provided by support.                                                                                                                                                                                                                                                                                                                                |
| Microsoft | Adjustment                         | Adjustment | Rounding errors.                                                                                                                                                                                                                                                                                                                                           |
| Microsoft | Tax                                | Tax        | US sales tax or VAT.                                                                                                                                                                                                                                                                                                                                       |
```

File: supporting_content/columns/chargecategory.md [hunk 10]
```diff
@@ -8,111 +8,111 @@ Current column mappings found in available data sets:
...
+| :------- | :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Refund   | Any adjustments that are applied after the original usage or purchase row. Adjustments may be related to multiple charges. (NOTE: Tax excluded)                               |
+| Credit   | Credits assoicated with promotional usage or incentives                                                                                                                       |
+| Purchase | Charges for the acquisition of a service or resource bought upfront or on a recurring basis.                                                                                  |
+| Tax      | Applicable taxes that are levied by the relevant authorities. Tax charges may vary depending on factors such as the location, jurisdiction, and local or federal regulations. |
+| Usage    | Charges based on the quantity of a service or resource that was consumed over a given period of time.                                                                         |
```
Comment (flanakin) [inline supporting_content/columns/chargecategory.md:50]:
```suggestion
| Value      | Description                          |
| :--------- | :------------------------------------|
| Refund     | Any adjustments that are applied after the original usage or purchase row. Adjustments may be related to multiple charges. (NOTE: Tax excluded)   |
| Credit     | Credits assoicated with promotional usage or incentives   |
| Purchase   | Charges for the acquisition of a service or resource bought upfront or on a recurring basis.              |
| Tax        | Applicable taxes that are levied by the relevant authorities. Tax charges may vary depending on factors such as the location, jurisdiction, and local or federal regulations. |
| Usage      | Charges based on the quantity of a service or resource that was consumed over a given period of time.     |
```

File: supporting_content/columns/chargecategory.md [hunk 11]
```diff
@@ -8,111 +8,111 @@ Current column mappings found in available data sets:
...
-| Usage      | Charges based on the quantity of a service or resource that was consumed over a given period of time.     |
+| Value    | Description                                                                                                                                                                   |
+| :------- | :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Purchase | Charges for the acquisition of a service or resource bought upfront or on a recurring basis.                                                                                  |
+| Tax      | Applicable taxes that are levied by the relevant authorities. Tax charges may vary depending on factors such as the location, jurisdiction, and local or federal regulations. |
+| Usage    | Charges based on the quantity of a service or resource that was consumed over a given period of time.                                                                         |
```
Comment (flanakin) [inline supporting_content/columns/chargecategory.md:60]:
```suggestion
| Value      | Description                          |
| :--------- | :------------------------------------|
| Purchase   | Charges for the acquisition of a service or resource bought upfront or on a recurring basis.              |
| Tax        | Applicable taxes that are levied by the relevant authorities. Tax charges may vary depending on factors such as the location, jurisdiction, and local or federal regulations. |
| Usage      | Charges based on the quantity of a service or resource that was consumed over a given period of time.     |
```

File: supporting_content/columns/chargecategory.md [hunk 12]
```diff
@@ -8,111 +8,111 @@ Current column mappings found in available data sets:
...
-| Value      | Description                          |
-| :--------- | :------------------------------------|
-| NULL       | Default value for all incomming charges.             |
-| Refund     | Refunded related to usage or purchase specific activities (expects a matching 'tax' transaction) |
-| Credit     | Promotional / negotiated / incentive credits provided at providers discression (does NOT expect a matching 'tax' transaction)       |
+| Value  | Description                                                                                                                   |
+| :----- | :---------------------------------------------------------------------------------------------------------------------------- |
+| NULL   | Default value for all incomming charges.                                                                                      |
+| Refund | Refunded related to usage or purchase specific activities (expects a matching 'tax' transaction)                              |
+| Credit | Promotional / negotiated / incentive credits provided at providers discression (does NOT expect a matching 'tax' transaction) |
```
Comment (flanakin) [inline supporting_content/columns/chargecategory.md:69]:
```suggestion
| Value      | Description                          |
| :--------- | :------------------------------------|
| NULL       | Default value for all incomming charges.             |
| Refund     | Refunded related to usage or purchase specific activities (expects a matching 'tax' transaction) |
| Credit     | Promotional / negotiated / incentive credits provided at providers discression (does NOT expect a matching 'tax' transaction)       |
```

File: supporting_content/columns/chargecategory.md [hunk 13]
```diff
@@ -8,111 +8,111 @@ Current column mappings found in available data sets:
...
+| :-------------- | :------------------ | :------------------------------------------- |
+| Usage           | NULL                | general usage                                |
+| Usage           | Refund              | service specific refunds /..e.g miss billing |
+| Usage           | Credit              | service specific incentives                  |
+| Purchase        | NULL                | general marketplace or 3rd party purchase    |
+| Purchase        | Refund              | bulk / general refunds                       |
+| Purchase        | Credit              | Non-service / usage specific credits         |
+| Tax             | NULL                | general tax                                  |
+| Tax             | Refund              | tax refund for usage or purchase refunded    |
+| Tax             | Credit              | NOT APPLICABLE                               |
```
Comment (flanakin) [inline supporting_content/columns/chargecategory.md:83]:
```suggestion
| Charge Category     | Adjustment Category         | Example usage                 |
| :--------- | :------------------------------------|:------------------------------|
|Usage       | NULL                                 | general usage                                                                                |
|Usage       | Refund                               | service specific refunds /..e.g miss billing                                                 |
|Usage       | Credit                               | service specific incentives                                                                  |
|Purchase    | NULL                                 | general marketplace or 3rd party purchase                                                    |
|Purchase    | Refund                               | bulk / general refunds                                                                       |
|Purchase    | Credit                               | Non-service / usage specific credits                                                         |
|Tax         | NULL                                 | general tax                                                                                  |
|Tax         | Refund                               | tax refund for usage or purchase refunded                                                    |
|Tax         | Credit                               | NOT APPLICABLE                                                                               |
```

File: supporting_content/columns/chargecategory.md [hunk 14]
```diff
@@ -8,111 +8,111 @@ Current column mappings found in available data sets:
...
+| Value    | Description                                                                                                                                                                   |
+| :------- | :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Credit   | Credits assoicated with promotional usage or incentives                                                                                                                       |
+| Purchase | Charges for the acquisition of a service or resource bought upfront or on a recurring basis.                                                                                  |
+| Tax      | Applicable taxes that are levied by the relevant authorities. Tax charges may vary depending on factors such as the location, jurisdiction, and local or federal regulations. |
+| Usage    | Charges based on the quantity of a service or resource that was consumed over a given period of time.                                                                         |
```
Comment (flanakin) [inline supporting_content/columns/chargecategory.md:92]:
```suggestion
| Value      | Description                          |
| :--------- | :------------------------------------|
| Credit     | Credits assoicated with promotional usage or incentives   |
| Purchase   | Charges for the acquisition of a service or resource bought upfront or on a recurring basis.              |
| Tax        | Applicable taxes that are levied by the relevant authorities. Tax charges may vary depending on factors such as the location, jurisdiction, and local or federal regulations. |
| Usage      | Charges based on the quantity of a service or resource that was consumed over a given period of time.     |
```

File: supporting_content/columns/chargecategory.md [hunk 15]
```diff
@@ -8,111 +8,111 @@ Current column mappings found in available data sets:
...
-| Bulk Refund     | General refund (expects a matching 'tax' transaction) |
-| Rounding Error     | Small corrections - Applicable to current billing period only |
-| Other    | Catch all       |
+| Value          | Description                                                                                      |
+| :------------- | :----------------------------------------------------------------------------------------------- |
+| NULL           | Default value for all incomming charges.                                                         |
+| Refund         | Refunded related to usage or purchase specific activities (expects a matching 'tax' transaction) |
+| Bulk Refund    | General refund (expects a matching 'tax' transaction)                                            |
+| Rounding Error | Small corrections - Applicable to current billing period only                                    |
+| Other          | Catch all                                                                                        |
```
Comment (flanakin) [inline supporting_content/columns/chargecategory.md:102]:
```suggestion
| Value      | Description                          |
| :--------- | :------------------------------------|
| NULL       | Default value for all incomming charges.             |
| Refund     | Refunded related to usage or purchase specific activities (expects a matching 'tax' transaction) |
| Bulk Refund     | General refund (expects a matching 'tax' transaction) |
| Rounding Error     | Small corrections - Applicable to current billing period only |
| Other    | Catch all       |
```

File: supporting_content/columns/chargecategory.md [hunk 16]
```diff
@@ -8,111 +8,111 @@ Current column mappings found in available data sets:
...
+| Charge Category | Adjustment Category | Example usage                                  |
+| :-------------- | :------------------ | :--------------------------------------------- |
+| Usage           | NULL                | general usage                                  |
+| Usage           | Refund              | service specific refunds /..e.g miss billing   |
+| Purchase        | NULL                | general marketplace or 3rd party purchase      |
+| Purchase        | Refund              | specific / 3rd party refund                    |
+| Tax             | NULL                | general tax                                    |
+| Tax             | Refund              | tax refund for usage or purchase refunded      |
+| Credit          | NULL                | service specific incentives                    |
+| Credit          | Other               | vendor specific / non usage related incentives |
```
Comment (flanakin) [inline supporting_content/columns/chargecategory.md:115]:
```suggestion
| Charge Category     | Adjustment Category         | Example usage                 |
| :--------- | :------------------------------------|:------------------------------|
|Usage       | NULL                                 | general usage                                                                                |
|Usage       | Refund                               | service specific refunds /..e.g miss billing                                                 |
|Purchase    | NULL                                 | general marketplace or 3rd party purchase                                                    |
|Purchase    | Refund                               | specific / 3rd party refund                                                                  |
|Tax         | NULL                                 | general tax                                                                                  |
|Tax         | Refund                               | tax refund for usage or purchase refunded                                                    |
|Credit      | NULL                                 | service specific incentives                                                                  |
|Credit      | Other                                | vendor specific / non usage related incentives                                               |
```

File: supporting_content/columns/chargecategory.md [hunk 17]
```diff
@@ -149,12 +149,12 @@ Permutations:
...
-| Usage invoice tax charge   | Tax   |  NULL |  NULL | Recurring        |
+| Scenario                                                     | ChargeCategory | PricingCategory  | Charge Frequency |
+| ------------------------------------------------------------ | -------------- | ---------------- | ---------------- |
+| Upfront discount purchase                                    | Purchase       | On-Demand        | One-time         |
+| Partial Upfront discount monthly fee                         | Purchase       | On-Demand        | Recurring        |
+| Usage covered by upfront portion of partial upfront discount | Usage          | Commitment-based | Usage-based      |
+| Unused commitment of partial upfront discount                | Usage          | Commitment-based | Usage-based      |
+| Usage not covered by discount                                | Usage          | On-Demand        | Usage-based      |
+| Refund                                                       | Adjustment     | NULL             | One-time         |
+| Usage invoice tax charge                                     | Tax            | NULL             | Recurring        |
```
Comment (flanakin) [inline supporting_content/columns/chargecategory.md:160]:
```suggestion
| Scenario| ChargeCategory | ChargeSubcategory | PricingCategory  | Charge Frequency |
|-|-|-|-|-|
| Upfront discount purchase | Purchase| NULL  | On-Demand  | One-time |
| Partial Upfront discount monthly fee | Purchase    | NULL  | On-Demand  | Recurring  |
| Usage covered by upfront portion of partial upfront discount | Usage   | Used Commitment   | Commitment-based | Usage-based  |
| Unused commitment of partial upfront discount    | Usage   | Unused Commitment | Commitment-based | Usage-based      |
| Usage not covered by discount | Usage | On-Demand  | On-Demand   | Usage-based|
| Refund | Adjustment   | Refund | NULL | One-time  |
| Usage invoice tax charge   | Tax   |  NULL |  NULL | Recurring        |
```

**Review by udam-f2 — APPROVED** [review]

**Review by rileyjenk — APPROVED** [review]

**Review by ijurica — APPROVED** [review]

**Review by aqu-erp — APPROVED** [review]

**Pull request merged**